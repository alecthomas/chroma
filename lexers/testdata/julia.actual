% Defining basic types

struct Value
    v::Int
end
const r = 33

A = zeros(3,3)
B = rand(4,3)

type = typeof(B)
C = A * B'
res = C' .+ r

const abstract = 33
abstract type K end

primitive type MyFloat32 32 end
