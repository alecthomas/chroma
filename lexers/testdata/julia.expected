[
  {"type":"Operator","value":"%"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"Defining"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"basic"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"types"},
  {"type":"Text","value":"\n\n"},
  {"type":"Keyword","value":"struct"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"Value"},
  {"type":"Text","value":"\n    "},
  {"type":"Name","value":"v"},
  {"type":"Operator","value":"::"},
  {"type":"KeywordType","value":"Int"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"const"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"r"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"33"},
  {"type":"Text","value":"\n\n"},
  {"type":"Name","value":"A"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"zeros"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumberInteger","value":"3"},
  {"type":"Punctuation","value":","},
  {"type":"LiteralNumberInteger","value":"3"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n"},
  {"type":"Name","value":"B"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"rand"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumberInteger","value":"4"},
  {"type":"Punctuation","value":","},
  {"type":"LiteralNumberInteger","value":"3"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n\n"},
  {"type":"Name","value":"type"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"typeof"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"B"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n"},
  {"type":"Name","value":"C"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"A"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"*"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"B"},
  {"type":"Operator","value":"'"},
  {"type":"Text","value":"\n"},
  {"type":"Name","value":"res"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"Name","value":"C"},
  {"type":"Operator","value":"'"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":".+"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"r"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"const"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"abstract"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"33"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"abstract type"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"K"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n\n"},
  {"type":"Keyword","value":"primitive type"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"MyFloat32"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"32"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n"}
]
